packages/core/src/__tests__/document.test.ts

```diff
@@ -277,35 +277,76 @@ describe('Document', () => {
       expect(doc.state.signature).toEqual(SignatureStatus.SIGNED)
       expect(doc.state.anchorStatus).not.toEqual(AnchorStatus.NOT_REQUESTED)
 
+      // Apply a final record that never gets anchored and thus never becomes a proper version
+      const finalContent = {foo: 'bar'}
+      const updateRec2 = await TileDoctype._makeRecord(doc.doctype, user, finalContent, doc.controllers)
+      await doc.applyRecord(updateRec2)
+
+      versions = doc.allVersionIds
+      expect(versions.length).toEqual(3)
+      expect(doc.versionId).toEqual(version2)
+      expect(doc.state.log.length).toEqual(5)
+
       // try to checkout non-existing version
       try {
-        await Document.loadVersion(doc, new CID('bafybeig6xv5nwphfmvcnektpnojts33jqcuam7bmye2pb54adnrtccjlsu'))
+        await doc.loadVersion(new CID('bafybeig6xv5nwphfmvcnektpnojts33jqcuam7bmye2pb54adnrtccjlsu'))
         throw new Error('Should not be able to fetch non-existing version')
       } catch (e) {
-        expect(e.message).toContain('No record found for version')
+        expect(e.message).toContain('No record found for CID')
       }
 
-      // try to checkout not anchored version
-      try {
-        await Document.loadVersion(doc, doc.doctype.state.log[2].cid)
-        throw new Error('Should not be able to fetch not anchored version')
-      } catch (e) {
-        expect(e.message).toContain('No anchor record for version')
-      }
-
-      const docV1 = await Document.loadVersion(doc, doc.doctype.state.log[1].cid)
+      // Correctly check out a specific version
+      const version = doc.doctype.state.log[1].cid
+      const versionId = DocID.fromOther(doc.id, version)
+      const docV1 = await Document.load(versionId, doctypeHandler, dispatcher, pinStore, context)
       expect(docV1.state.log.length).toEqual(2)
       expect(docV1.controllers).toEqual(controllers)
       expect(docV1.content).toEqual(initialContent)
       expect(docV1.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
 
-      // try to call doctype.change
+      // try to call doctype.change on doc that's tied to a specific version
       try {
         await docV1.doctype.change({ content: doc.content, controllers: doc.controllers })
-        throw new Error('Should not be able to fetch not anchored version')
+        throw new Error('Should not be able to change document that was loaded at a specific version')
```

Comment:
can we use `jest`s `fail` here: `fail('Should not be able to change document that was loaded at a specific version')`

Reply:
yep, there are more places in the codebase definitely where we can change that